src/sql/workbench/parts/notebook/notebookActions.ts

```diff
@@ -158,15 +164,25 @@ export class IMultiStateData<T> {
 
 export abstract class MultiStateAction<T> extends Action {
 
-	constructor(id: string, protected states: IMultiStateData<T>) {
+	constructor(id: string, protected states: IMultiStateData<T>, private _keybindingService: IKeybindingService) {
 		super(id, '');
 		this.updateLabelAndIcon();
 	}
 
 	private updateLabelAndIcon() {
-		this.label = this.states.label;
-		this.tooltip = this.states.tooltip;
-		this.class = this.states.classes;
+		try {
+			// If a keyboard shortcut exists for the command id passed in, append that to the label
+			let keyboardShortcut: string;
+			if (this.states.commandId !== '') {
+				let binding = this._keybindingService.lookupKeybinding(this.states.commandId);
+				keyboardShortcut = binding ? binding.getLabel() : undefined;
+			}
+			this.label = this.states.label;
```

Comment:
>this.label = this.states.label; [](start = 3, length = 31)

How about setting the label and class first so that those will always be done even if the lookup throws an error (which is why I assume you added the try/catch). The point being to ensure that existing behavior isn't modified - it'll be able to show the label and icon as expected even if the keybinding tooltip part doesn't happen